Packages/com.unity.inputsystem/InputSystem/Editor/AssetEditor/InputActionPropertiesView.cs

```diff
@@ -47,10 +47,6 @@ protected override void DrawGeneralProperties()
 
             if (EditorGUI.EndChangeCheck())
             {
-                // When changing the action type, clear out the expected control type.
-                if ((InputActionType)m_SelectedActionType != (InputActionType)m_ActionTypeProperty.intValue)
-                    m_SelectedControlType = 0;
```

Comment:
I think we need to replace this with something else rather than removing it. Problem is that when you change from, say, Value of Vector2 to Button, then the Vector2 constraint should indeed be gotten rid of I think. Probably makes sense to restrict this to that particular direction (anything -> Button), though.

Reply:
I think that is still guaranteed by the line below:

if ((InputActionType)m_SelectedActionType == InputActionType.Button)
                    m_ExpectedControlTypeProperty.stringValue = "Button";

Reply:
Oh, you're right. Good catch.